lms/djangoapps/grades/tasks.py

```diff
@@ -315,14 +318,21 @@ def _update_subsection_grades(
     student = User.objects.get(id=user_id)
     store = modulestore()
     with store.bulk_operations(course_key):
-        course_structure = get_course_blocks(student, store.make_course_usage_key(course_key))
+        course_usage_key = store.make_course_usage_key(course_key)
+        course_structure = get_course_blocks(student, course_usage_key)
         subsections_to_update = course_structure.get_transformer_block_field(
             scored_block_usage_key,
             GradesTransformer,
             'subsections',
             set(),
         )
 
+        # Clear the course cache if access is restricted and course blocks are
+        # cached without the restricted blocks.
+        if not subsections_to_update:
+            clear_course_from_cache(course_usage_key.course_key)
+            raise BlockStructureNotFound(scored_block_usage_key)
```

Comment:
I think you should create a new exception named `SubsectionNotFound` [here](https://github.com/openedx/edx-platform/blob/b4552850c50da3eb3230fcc120c835a516e4644a/lms/djangoapps/grades/exceptions.py) instead of raising `BlockStructureNotFound` because it appears block structure is available but few blocks are missing.

Reply:
Can I use [`UsageKeyNotInBlockStructure`](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content/block_structure/exceptions.py#L20)?

Reply:
Yes that can be used too.